Looks like it exposed `iwd`'s incomplete header inclusion (or similar):

```
error: builder for '/nix/store/31yjgscir510w00a129l5kj10g4dgh4m-iwd-2.8.drv' failed with exit code 2;
       last 10 log lines:
       > src/ip-pool.c:337:34: error: invalid use of undefined type 'const struct ifaddrmsg'
       >   337 |                 rec.ifindex = ifa->ifa_index;
       >       |                                  ^~
       > src/ip-pool.c: In function 'ip_pool_init':
       > src/ip-pool.c:366:39: error: 'RTNLGRP_IPV4_IFADDR' undeclared (first use in this function)
       >   366 |         if (!l_netlink_register(rtnl, RTNLGRP_IPV4_IFADDR,
       >       |                                       ^~~~~~~~~~~~~~~~~~~
       > make[1]: *** [Makefile:2538: src/ip-pool.o] Error 1 shuffle=3868977690
       > make[1]: *** Waiting for unfinished jobs....
       > make: *** [Makefile:1710: all] Error 2 shuffle=3868977690
       For full logs, run 'nix log /nix/store/31yjgscir510w00a129l5kj10g4dgh4m-iwd-2.8.drv'.
```

--- a/pkgs/os-specific/linux/ell/default.nix
+++ b/pkgs/os-specific/linux/ell/default.nix
@@ -9,14 +9,14 @@
 
 stdenv.mkDerivation rec {
   pname = "ell";
-  version = "0.59";
+  version = "0.61";
 
   outputs = [ "out" "dev" ];
 
   src = fetchgit {
     url = "https://git.kernel.org/pub/scm/libs/ell/ell.git";
     rev = version;
-    hash = "sha256-uJcGYT+JSdz/XTyJb/VUyedmSKJW/4BbTM3fw3ebtIc=";
+    hash = "sha256-spoZRT/gBCk/e/pn1AujCpCPdEM7hn/ImRyQq4hwctI=";
   };
 
   nativeBuildInputs = [

--- a/pkgs/os-specific/linux/iwd/default.nix
+++ b/pkgs/os-specific/linux/iwd/default.nix
@@ -1,5 +1,6 @@
 { lib, stdenv
 , fetchgit
+, fetchpatch
 , autoreconfHook
 , pkg-config
 , ell
@@ -19,6 +20,16 @@ stdenv.mkDerivation rec {
     rev = version;
     sha256 = "sha256-i+2R8smgLXooApj0Z5e03FybhYgw1X/kIsJkrDzW8y4=";
   };
+
+  patches = [
+    # ell-0.61 compatibility:
+    # TODO: remove with update to 2.10
+    (fetchpatch {
+      name = "rtnetlink-headers.patch";
+      url = "https://git.kernel.org/pub/scm/network/wireless/iwd.git/patch/?id=653122498aaf660d3bf9263bf95aa6544ef154c6";
+      hash = "sha256-z+TSWwJkSZm9IbKnJUHwcGannhgeiCiPurh/0a8TwKA=";
+    })
+  ];
 
   outputs = [ "out" "man" "doc" ]
     ++ lib.optional (stdenv.hostPlatform == stdenv.buildPlatform) "test";

--- a/pkgs/os-specific/linux/iwd/default.nix
+++ b/pkgs/os-specific/linux/iwd/default.nix
@@ -1,6 +1,5 @@
 { lib, stdenv
 , fetchgit
-, fetchpatch
 , autoreconfHook
 , pkg-config
 , ell
@@ -9,27 +8,18 @@
 , readline
 , openssl
 , python3Packages
+, gitUpdater
 }:
 
 stdenv.mkDerivation rec {
   pname = "iwd";
-  version = "2.8";
+  version = "2.10";
 
   src = fetchgit {
     url = "https://git.kernel.org/pub/scm/network/wireless/iwd.git";
     rev = version;
-    sha256 = "sha256-i+2R8smgLXooApj0Z5e03FybhYgw1X/kIsJkrDzW8y4=";
+    hash = "sha256-zePFmcQRFjcH6KToTpBFMQzGY+Eq7jijfn0R/MMKGrw=";
   };
-
-  patches = [
-    # ell-0.61 compatibility:
-    # TODO: remove with update to 2.10
-    (fetchpatch {
-      name = "rtnetlink-headers.patch";
-      url = "https://git.kernel.org/pub/scm/network/wireless/iwd.git/patch/?id=653122498aaf660d3bf9263bf95aa6544ef154c6";
-      hash = "sha256-z+TSWwJkSZm9IbKnJUHwcGannhgeiCiPurh/0a8TwKA=";
-    })
-  ];
 
   outputs = [ "out" "man" "doc" ]
     ++ lib.optional (stdenv.hostPlatform == stdenv.buildPlatform) "test";
@@ -97,6 +87,11 @@ stdenv.mkDerivation rec {
   '';
 
   enableParallelBuilding = true;
+
+  passthru.updateScript = gitUpdater {
+    # No nicer place to find latest release.
+    url = "https://git.kernel.org/pub/scm/network/wireless/iwd.git";
+  };
 
   meta = with lib; {
     homepage = "https://git.kernel.org/pub/scm/network/wireless/iwd.git";

